shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/test/java/org/apache/shardingsphere/sql/parser/integrate/asserts/segment/JoinSpecificationAssert.java

```diff
@@ -37,5 +64,142 @@
      * @param expected expected JoinSpecification
      */
     public static void assertIs(final SQLCaseAssertContext assertContext, final JoinSpecificationSegment actual, final ExpectedJoinSpecification expected) {
+        assertThat(assertContext.getText("startindex assert error"), actual.getStartIndex(), is(expected.getStartIndex()));
+        assertThat(assertContext.getText("stopindex assert error"), actual.getStopIndex(), is(expected.getStopIndex()));
+        assertAndPredicates(assertContext, actual.getAndPredicates(), expected.getAndPredicates());
+    
+    }
+    
+    private static void assertAndPredicates(final SQLCaseAssertContext assertContext, final Collection<AndPredicate> actual, final List<ExpectedAndPredicate> expected) {
```

Comment:
Those functions are the same as these in `WhereClauseAssert`. My suggestion is to extract `PredicatesAssert`, which will be used by `WhereClauseAssert` and this one.

Reply:
Ok.